packages/data-service-generator/src/server/create-main/create-main-file.ts

```diff
@@ -1,38 +1,50 @@
 import path from "path";
-import { promises as fs } from "fs";
-import { Module, ModuleMap } from "@amplication/code-gen-types";
+import {
+  CreateMainFileParams,
+  EventNames,
+  Module,
+  ModuleMap,
+} from "@amplication/code-gen-types";
 import { MAIN_TS_FILE_NAME, MAIN_TS_WITH_BIGINT_FILE_NAME } from "../constants";
 import DsgContext from "../../dsg-context";
-import { formatCode } from "@amplication/code-gen-utils";
-
-export async function createMainFile() {
-  const { logger, serverDirectories, hasBigIntFields } = DsgContext.getInstance;
-  const moduleMap = new ModuleMap(logger);
-
-  await logger.info("Creating main.ts file...");
+import { formatCode, print, readFile } from "@amplication/code-gen-utils";
+import pluginWrapper from "../../plugin-wrapper";
 
+export async function createMainFile(): Promise<ModuleMap> {
   const mainFilePath = path.resolve(__dirname, MAIN_TS_FILE_NAME);
   const mainWithBigintFilePath = path.resolve(
     __dirname,
     MAIN_TS_WITH_BIGINT_FILE_NAME
   );
-  const mainFileContent = await fs.readFile(mainFilePath, "utf-8");
-  const mainWithBigintFileContent = await fs.readFile(
-    mainWithBigintFilePath,
-    "utf-8"
-  );
+  const template = await readFile(mainFilePath);
+  const bigIntTemplate = await readFile(mainWithBigintFilePath);
```

Comment:
I'd recommend to move the logic that handle the template selection before so the plugin event would need to only take care of one template, making it simplier and more maintainable. @Shurtu-gal what do you think?

Reply:
I could do that, however there is a difference between the two, which one should the template pass ?

diff: https://github.com/amplication/amplication/blob/2e4118c079dfc69b516598b843ec97fc7310ccfa/packages/data-service-generator/src/server/create-main/main.with-bigint.template.ts#L26-L31

Reply:
we can unify the two with the following patch 
[refactor-main.ts.patch](https://github.com/amplication/amplication/files/14981999/refactor-main.ts.patch)

Reply:
So basically you want this to be applied in the AST tree, if I am able to understand it correctly.

Reply:
Done @overbit.